### PR TITLE
Updating documentation for Microsoft.VisualStudio with project resour…

### PR DESCRIPTION
**Pull request by niblak**
…ce APIs. Corrections to existing examples.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/visualstudio/resource-manager/Microsoft.VisualStudio/2014-04-01-preview/examples/CreateProjectResource.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿{
+  "parameters": {
+    "body": {
+        "id": "/subscriptions/0de7f055-dbea-498d-8e9e-da287eedca90/resourceGroups/VS-Example-Group/providers/Microsoft.VisualStudio/account/ExampleAccount/project/ExampleProject",
+        "name": "ExampleProject",
+        "type": "Microsoft.VisualStudio/account/project",
+        "location": "Central US",
+        "tags": {},
+        "properties": {
+          "ProcessTemplateId": "6B724908-EF14-45CF-84F8-768B5384DA45",
```
Comment (niblak) [inline specification/visualstudio/resource-manager/Microsoft.VisualStudio/2014-04-01-preview/examples/CreateProjectResource.json]: I am not sure how best to express the fact that these two properties are required (ProcessTemplateId and VersionControlOption).

Also, there's a relatively static range of values that are accepted for these. If they were formal parameters I'd consider using x-ms-enum in the RP spec to list all possible values, but in this particular case I am not sure how to express that in documentation.

File: specification/visualstudio/resource-manager/Microsoft.VisualStudio/2014-04-01-preview/Csm.json [hunk 2]
```diff
@@ -789,6 +1072,38 @@
...
+    },
+    "ProjectResource": {
+      "description": "A Visual Studio Team Services project resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
```
Comment (balajikris) [inline specification/visualstudio/resource-manager/Microsoft.VisualStudio/2014-04-01-preview/Csm.json]: you could model this like so:
```json
       "properties": {
         "properties": {
           "ProcessTemplateId": { "required": true },
           "VersionControlOption": { "required": true },
           "properties": {
               "type": "object",
    	       "additionalProperties": {
			"type": "string"
		 },
		 "x-ms-client-flatten": true
           },
          "x-ms-client-flatten": true
         }
       }
```

File: specification/visualstudio/resource-manager/Microsoft.VisualStudio/2014-04-01-preview/examples/CreateProjectResource.json (hunk 1 above)
Comment (balajikris) [inline specification/visualstudio/resource-manager/Microsoft.VisualStudio/2014-04-01-preview/examples/CreateProjectResource.json]: I made a comment about this [here](https://github.com/Azure/azure-rest-api-specs/pull/1498/files#diff-3f2a2429b806e0c507ad43a0e8aefc09R1084). I think that should work, but you could quickly verify that by generating C# and viewing the definition for `ProjectResource`

Comment (niblak) [inline specification/visualstudio/resource-manager/Microsoft.VisualStudio/2014-04-01-preview/examples/CreateProjectResource.json]: Thanks, I'll give that a try.

Is it possible to use the `enum` semantics on non-string types? ProcessTemplateId is actually a GUID, and the only accepted values are 6B724908-EF14-45CF-84F8-768B5384DA45, ADCC42AB-9882-485E-A3ED-7678F01F66BC, 27450541-8E31-4150-9947-DC59F998FC01. 

From looking at other RP specs and the Swagger docs, I don't see a way to assign a friendly name to these (the IDs correspond to Scrum, Agile and Cmmi process templates).

Comment (balajikris) [inline specification/visualstudio/resource-manager/Microsoft.VisualStudio/2014-04-01-preview/examples/CreateProjectResource.json]: you are correct that there isn't a way to do this as of today. For the future, I hear that there is a proposal to expand the scope of [x-ms-enum](https://github.com/Azure/azure-rest-api-specs/blob/dce4da0d748565efd2ab97a43d0683c2979a974a/documentation/swagger-extensions.md#x-ms-enum) to accommodate this kind of friendly name assignment.

**Comment by balajikris** [comment]
@niblak - The CI job found some errors with the new examples you added for the operations that are being added by this PR. See [here](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/259997568)
If you scroll all the way to the end, you can see that the last 4-5 failing examples were the ones you added. Could you please update your sample to make validation pass?

**Comment by balajikris** [comment]
@niblak -- I can see you are working on this, can you please tag me in a comment after you are done fixing the validation errors? Thanks.

**Comment by niblak** [comment]
@balajikris sorry about all the pushes.

I've resolved most of the example errors but am confused on the remaining two: INVALID_RESPONSE_BODY and INVALID_REQUEST_PARAMETER. 

What am I missing here? It seems to be complaining about properties that are actually part of the response/request body for our API.

[Latest PR build here.](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/262352792)

File: specification/visualstudio/resource-manager/Microsoft.VisualStudio/2014-04-01-preview/examples/UpdateProjectResource.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+﻿{
+  "parameters": {
+    "body": {
+      "id": "/subscriptions/0de7f055-dbea-498d-8e9e-da287eedca90/resourceGroups/VS-Example-Group/providers/microsoft.visualstudio/account/ExampleAccount/project/ExampleProject",
+      "name": "ms.example",
+      "type": "Microsoft.VisualStudio/account/extension",
+      "location": "Central US",
+      "tags": {},
+      "properties": {},
+      "plan": {
```
Comment (balajikris) [inline specification/visualstudio/resource-manager/Microsoft.VisualStudio/2014-04-01-preview/examples/UpdateProjectResource.json]: This should be removed

**Comment by balajikris** [comment]
Hey @niblak -

Sure, Its complaining because you have `plan` as an input to your `updateproject` call but the schema doesn't say anything about `plan` being an input to the api call. This is probably a result of copy pasting from another example? For e.g. see `/UpdateExtensionResource.json` example, `plan` is one of the inputs but in this case, the spec defines it as one of the input params.

**Comment by niblak** [comment]
Thanks @balajikris. That was the problem and the CI build is passing now. Build #10 is failing due to a null ref, should I be concerned about this one?

The syntax for specifying ProcessTemplateId and VersionControlOption as required didn't seem to work unfortunately (failed both the PR build and C# client generation). For now, I've just added the list of valid values to the method summary. Not ideal, but there seems to be precedent for this in other RP specs.

**Comment by balajikris** [comment]
@niblak  - don't worry about the null ref on the build. Seems like a bug in the tool. CI looks much nicer now, thanks for making hte changes 👍 Will review again in a bit

**Review by balajikris — APPROVED** [review]

**Comment by balajikris** [comment]
@niblak - If you are all done here, I'm ready to pull this in. Let me know. Thanks.

**Comment by niblak** [comment]
@balajikris Good to go. Thanks for the review!

**Pull request merged**